win32ss/drivers/displays/vga/objects/screen.c

```diff
@@ -139,7 +139,7 @@ BOOL InitVGA(PPDEV ppdev, BOOL bFirst)
 
     ppdev->sizeSurf.cx = 640;
     ppdev->sizeSurf.cy = 480;
-    ppdev->ModeNum = 12;
+    ppdev->ModeNum = 2;
```

Comment:
You mention that vga_new says this is mode "2", and that's why you're adjusting it there.
However, do you *know* why vga_new reports this as mode "2" instead of "12" ?

Reply:
My thought is, maybe `12` was just a typo. If I remember correctly, mode `1` is for monochrome, and `2` for 16-color.

Reply:
In vga_new/vgadata.c, the 640x480x16 mode entry is the third one of ModesVGA array, ie index 2.

Reply:
Ah, it's an index in some array, not a mode number like those coming from the BIOS. OK